### PR TITLE
Refactor some parts of contract and accounts packages

### LINKED ISSUE #7196: Refactor some parts of Contract, Accounts and Eth packages
Refactor some parts of Contract, Accounts and Eth packages to be more usable inside web3.js ZKSync plugin .

### PR DESCRIPTION
**Pull request by Muhammad-Altabba**
## Description

Fixes #7196

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] I have selected the correct base branch.
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [ ] My changes generate no new warnings.
- [ ] Any dependent changes have been merged and published in downstream modules.
- [ ] I ran `npm run lint` with success and extended the tests and types if necessary.
- [ ] I ran `npm run test:unit` with success.
- [ ] I ran `npm run test:coverage` and my test cases cover all the lines and branches of the added code.
- [ ] I ran `npm run build` and tested `dist/web3.min.js` in a browser.
- [ ] I have tested my code on the live network.
- [ ] I have checked the Deploy Preview and it looks correct.
- [ ] I have updated the `CHANGELOG.md` file in the root folder.
- [ ] I have linked Issue(s) with this PR in "Linked Issues" menu.

### CODE REVIEW

File: packages/web3-eth-contract/src/contract.ts [hunk 1]
```diff
@@ -215,6 +218,182 @@ const contractSubscriptions = {
 	newBlockHeaders: NewHeadsSubscription,
 };
 
+
+/*
+ * This class is only supposed to be used for the return of `new Contract(...).deploy(...)` method.
+ */
+export class DeployerMethodClass<FullContractAbi extends ContractAbi> {
```
Comment (jdevcs) [inline packages/web3-eth-contract/src/contract.ts]: why there is need of moving `deploy` to different class?

Comment (jdevcs) [reply packages/web3-eth-contract/src/contract.ts]: (from usability perspective for external users)

Comment (Muhammad-Altabba) [reply packages/web3-eth-contract/src/contract.ts]: Good question,
I needed to add `populateTransaction`. But the `Contract` class was already about 1500 lines of code. So abstracting the logic into multiple classes will ease code maintenance for us. And if a user wanted to extend the `Contract` class he will have to deal with simpler code structures... 
And we would in the future also create a class for the return of the method `_createContractMethod` as it also returns a class like structure.

**Review by avkos — APPROVED** [review]

File: packages/web3-eth-contract/src/contract.ts [hunk 2]
```diff
@@ -15,6 +15,8 @@ You should have received a copy of the GNU Lesser General Public License
 along with web3.js.  If not, see <http://www.gnu.org/licenses/>.
 */
 
+/* eslint-disable max-classes-per-file */
```
Comment (jdevcs) [inline packages/web3-eth-contract/src/contract.ts]: I'll suggest to move new class in different file

Comment (jdevcs) [reply packages/web3-eth-contract/src/contract.ts]: LGTM just need above change

**Comment by avkos** [comment]
I've refactored plugin a little bit. now it is not necessary to register a new transaction type. so you can make changes in this PR too

**Comment by Muhammad-Altabba** [comment]
> I've refactored plugin a little bit. now it is not necessary to register a new transaction type. so you can make changes in this PR too

Thanks @avkos ,
The changes in this MR for the transaction type registration still fixes an issue that might happen if someone used the this functionality.

**Pull request merged**